core/store/orm/orm.go

```diff
@@ -1191,9 +1190,17 @@ func (orm *ORM) LogConsumptionExists(lc *models.LogConsumption) (bool, error) {
 	query := "SELECT id FROM log_consumptions " +
 		"WHERE block_hash=$1 " +
 		"AND log_index=$2 " +
-		"AND consumer_type=$3 " +
-		"AND consumer_id=$4"
-	return orm.rowExists(query, lc.BlockHash, lc.LogIndex, lc.ConsumerType, lc.ConsumerID)
+		"AND consumer_id=$3"
+	query = fmt.Sprintf("SELECT exists (%s)", query)
```

Comment:
This is a potential SQL injection attack vector.

Reply:
@sam nothing is interpolated into `query`. It's a constant string defined above. We could have have defined the whole string as one, but this I think reads cleaner. It's no more risky than string concatenation.

Reply:
I could do something like this..
```
query := "SELECT exists (" +
	"SELECT id FROM log_consumptions " +
	"WHERE block_hash=$1 " +
	"AND log_index=$2 " +
	"AND consumer_id=$3" +
	")"
```

but again they're equivalent operations and I think the former reads cleaner

Reply:
I think generally you shouldn't use `sprintf` anywhere near SQL even if your query is constant, because code changes over time and it could be easy for someone to accidentally add an injection attempt in, assuming that everyone was using this simple mantra. It's better to be safe than sorry.

Reply:
Fair enough. Removed!

Reply:
Thanks @RyanRHall